### PR TITLE
Fix RBAC for the APF controller

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Fixes the RBAC for the APF controller, necessary for the apiserver-extensions.

#### Which issue(s) this PR fixes:

Part of investigation for #1372 

#### Special notes for your reviewer:

I identified this problem while looking at the logs for the flaky E2E test.

This won't solve the flakiness, but will cleanup the log lines from one red-herring.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
It looks like there was a weird race condition trying to delete all pods after the test case finished.

/retest

**Comment by alculquicondor** [comment]
Let me see if I can catch a flake

/test pull-kueue-test-e2e-main-1-25
/test pull-kueue-test-e2e-main-1-26 
/test pull-kueue-test-e2e-main-1-27 
/test pull-kueue-test-e2e-main-1-28

**Comment by alculquicondor** [comment]
Yeah, #1372 is still there

/retest

File: Makefile [hunk 1]
```diff
@@ -118,7 +118,7 @@ update-helm: manifests yq
 	./hack/update-helm.sh
 
 .PHONY: generate
-generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations and client-go libraries.
+generate: gomod-download controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations and client-go libraries.
```
Comment (mimowo) [inline Makefile:121]: Is this related to the fix?

File: pkg/visibility/server.go [hunk 2]
```diff
@@ -90,6 +90,7 @@ func newVisibilityServerConfig() *genericapiserver.RecommendedConfig {
 	c.Config.OpenAPIV3Config.Info.Title = "Kueue visibility-server"
 	c.Config.OpenAPIConfig.Info.Version = strings.Split(c.Config.Version.String(), "-")[0]
 	c.Config.OpenAPIV3Config.Info.Version = strings.Split(c.Config.Version.String(), "-")[0]
+	c.FlowControl = nil
```
Comment (mimowo) [inline pkg/visibility/server.go:93]: Is it needed? Does it disable the P&F at the extension side?

Comment (alculquicondor) [reply pkg/visibility/server.go:93]: Oh, this was a different approach, removing.

File: Makefile (hunk 1 above)
Comment (alculquicondor) [reply Makefile:121]: No, but `generate` wasn't working for me without it.

Comment (mimowo) [reply Makefile:121]: sgtm

**Review by mimowo — COMMENTED** [review]
/lgtm

**Comment by tenzen-y** [comment]
Due to #1484 
/test pull-kueue-test-integration-main

**Review by tenzen-y — COMMENTED** [review]
/lgtm
/approve
